File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int func(vector<int> &nums, int k){
        int l = 0;
        int r = 0;
        int count = 0;
        int sum = 0;

        while(r < nums.size()){
            sum += nums[r] % 2;
            while(sum > k){
                sum -= nums[l] % 2;
                l++;
            }

            count+=r-l+1;
            r++;
        }
        return count;
    }

    int numberOfSubarrays(vector<int>& nums, int k) {
        return func(nums, k) - func(nums, k-1);
    }
};

File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        nums = [num % 2 for num in nums]
        # This now boils down to subarray with sum  = k 
        mapp = {}
        mapp[0] = 1
        summ = 0
        count = 0

        for i in range(len(nums)):
            summ += nums[i]
            remain = summ - k
            if remain in mapp:
                count += mapp[remain]

            mapp[summ] = mapp.get(summ, 0) + 1
        
        return count